client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/GetRollupJobResponse.java

```diff
@@ -0,0 +1,418 @@
+            builder.startObject();
+            builder.field(STATE.getPreferredName(), state.value());
+            if (currentPosition != null) {
+                builder.field(CURRENT_POSITION.getPreferredName(), currentPosition);
+            }
+            builder.field(UPGRADED_DOC_ID.getPreferredName(), upgradedDocumentId);
+            builder.endObject();
+            return builder;
+        }
+
+        private static final ConstructingObjectParser<RollupJobStatus, Void> PARSER = new ConstructingObjectParser<>(
+                STATUS.getPreferredName(),
+                true,
+                args -> {
+                    IndexerState state = (IndexerState) args[0];
+                    @SuppressWarnings("unchecked") // We're careful of the contents
+                    Map<String, Object> currentPosition = (Map<String, Object>) args[1];
+                    Boolean upgradedDocumentId = (Boolean) args[2];
+                    return new RollupJobStatus(state, currentPosition, upgradedDocumentId == null ? false : upgradedDocumentId);
+                });
+        static {
+            PARSER.declareField(constructorArg(), p -> IndexerState.fromString(p.text()), STATE, ObjectParser.ValueType.STRING);
+            PARSER.declareField(optionalConstructorArg(), p -> {
+                if (p.currentToken() == XContentParser.Token.START_OBJECT) {
+                    return p.map();
+                }
+                if (p.currentToken() == XContentParser.Token.VALUE_NULL) {
+                    return null;
+                }
+                throw new IllegalArgumentException("Unsupported token [" + p.currentToken() + "]");
+            }, CURRENT_POSITION, ObjectParser.ValueType.VALUE_OBJECT_ARRAY);
+
+            // Optional to accommodate old versions of state
+            PARSER.declareBoolean(ConstructingObjectParser.optionalConstructorArg(), UPGRADED_DOC_ID);
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            if (this == other) return true;
+            if (other == null || getClass() != other.getClass()) return false;
+            RollupJobStatus that = (RollupJobStatus) other;
+            return Objects.equals(state, that.state)
+                    && Objects.equals(currentPosition, that.currentPosition)
+                    && upgradedDocumentId == that.upgradedDocumentId;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(state, currentPosition, upgradedDocumentId);
+        }
+
+        @Override
+        public final String toString() {
+            return Strings.toString(this);
+        }
+    }
+
+    /**
+     * IndexerState represents the internal state of the indexer.  It
+     * is also persistent when changing from started/stopped in case the allocated
+     * task is restarted elsewhere.
+     */
+    public enum IndexerState {
```

Comment:
is this only valid in the context of a rollup job response? if not, can u make it so others can use it by making its own class.

Reply:
I believe it is only valid for this, yes.

Reply:
++ This is correct, atm only exposed to the user via the GetJob API

Reply:
:+1: